### PR TITLE
NVSHAS-8824: User fails to delete own groups, cannot create namespace-scoped groups

### PR DESCRIPTION
**Pull request by williamlin-suse**
Handle access denied error

### CODE REVIEW

File: controller/rest/group.go [hunk 1]
```diff
@@ -615,11 +615,14 @@ func handlerGroupCreate(w http.ResponseWriter, r *http.Request, ps httprouter.Pa
...
-	if exist, _ := cacher.DoesGroupExist(rg.Name, acc); exist {
+	if exist, err := cacher.DoesGroupExist(rg.Name, acc); exist {
 		e := "Group already exists"
 		log.WithFields(log.Fields{"name": rg.Name}).Error(e)
 		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrDuplicateName, e)
 		return
+	} else if err == common.ErrObjectAccessDenied {
+		restRespAccessDenied(w, login)
+		return
 	}
```
Comment (yasker) [inline controller/rest/group.go:626]: What about other errors?

**Review by yasker — CHANGES_REQUESTED** [review]

File: controller/rest/group.go (hunk 1 above)
Comment (williamlin-suse) [reply controller/rest/group.go:626]: There are 3 cases:
1. The to-be-created group exists & the caller has permission for that group -> handled by: if { ... }
2. The to-be-created group exists but the caller doesn't have permission for that group -> handled by: else if { ... }
3. The to-be-created does not exist -> the execution keeps going for creating the group

Comment (yasker) [reply controller/rest/group.go:626]: From reading the code, it is possible that `!exists && err != NotFound` exists. It is better not to rely on the assumption that the function won't return other errors. If it's unexpected, we need to at least log the error and return it.

Comment (williamlin-suse) [reply controller/rest/group.go:630]: code change as advised

**Review by jeffhuang4704 — APPROVED** [review]
looks good to me.

**Review by yasker — APPROVED** [review]

**Review by yasker — CHANGES_REQUESTED** [review]
LGTM. Please squash the commits locally.

**Review by yasker — APPROVED** [review]

**Pull request merged**